logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -741,7 +741,7 @@ class UserSessionScope internal constructor(
             messageDraftDAO = userStorage.database.messageDraftDAO,
         )
 
-    private val userRepository: UserRepository = UserDataSource(
+    private val userRepository: UserRepository get() = UserDataSource(
```

Comment:
i think this is initially a Singleton sice UserDataSource have a "userDetailsRefreshInstantCache"

Reply:
I missed that, you're right, I reverted that and moved some things around to make it work